### DISCUSSION TITLE
Documentation of incompatibilities with VBx?

### DISCUSSION THREAD

**Discussion by andrewhenry3** [twinbasic/twinbasic#1695] [Q&A] [answered]
Is there a master list of incompatibilities (by design or otherwise) that I should be checking before submitting Issues?  I see the two Issues stickies (Roadmap and Unimplemented properties).

**Comment by fafalone** [comment, marked as answer]
There's a separate unimplemented list for UCs here: #1157

Then obviously the major missing components are known... VBG files, ActiveX exes, MDI forms, Property Pages for in-project UCs, Printer object, and missing controls (only the OLE control from the basics, then several from common controls, and extended controls; Krool's VBCCR project works and has a 64bit compat tB version, which helps mitigate this). 

But for what is implemented, if it's not working it's a bug. Too many to go over really. Use the search feature in issues. Don't go too crazy, just do a basic check with common terms, nobody is going to jump down your throat for missing an already-reported issue ☺️

ETA: I'd also note what falls under the scope of '100% compatibility'. 100% compatibility is for the publicly documented language and COM. It doesn't include internal implementation details, so for instance the self-subclassing assembly thunks common in VB6 rely on undocumented internal structures, and won't be supported (however, tB supports self-subclass with regular methods natively: You can use `AddressOf` within a class, or to refer to class instance members from outside it). And for example The trick's multithreading module relies on undocumented calls to msvbm60.dll, so won't be supported (however, tB requires no hacks for that, it's thread-safe and you can call `CreateThread` without issue). But when VB objects implement publicly documented COM interfaces, like forms and `IOleInPlaceActiveObject`, those *are* supported, and some are implemented (that one is) but not all of them (e.g. IPerPropertyBrowsing not yet implemented)-- so if you find one not mentioned as known-missing, you should create an issue for it.
